src/Tizen.Security.WebAuthn/Tizen.Security.WebAuthn/AuthenticatorAssertionResponse.cs

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright (c) 2024 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License
+ */
+
+using static Interop;
+
+namespace Tizen.Security.WebAuthn
+{
+    /// <summary>
+    /// The response of GetAssertion().
+    /// </summary>
+    /// <remarks>
+    /// Refer to the following W3C specification for more information.
+    /// https://www.w3.org/TR/webauthn-3/#authenticatorassertionresponse
+    /// </remarks>
+    /// <since_tizen> 12 </since_tizen>
+    public class AuthenticatorAssertionResponse
+    {
+        internal AuthenticatorAssertionResponse(WauthnAuthenticatorAssertionResponse wauthnResponse)
+        {
+            ClientDataJson = NullSafeMarshal.PtrToArray(wauthnResponse.clientDataJson);
+            AuthenticatorData = NullSafeMarshal.PtrToArray(wauthnResponse.attestationObject);
+            Signature = NullSafeMarshal.PtrToArray(wauthnResponse.signature);
+            UserHandle = NullSafeMarshal.PtrToArray(wauthnResponse.userHandle);
+            AttestationObject = NullSafeMarshal.PtrToArray(wauthnResponse.attestationObject);
+        }
+
+        /// <summary>
+        /// JSON-compatible serialization of client data.
+        /// </summary>
+        public byte[] ClientDataJson { get; init; }
```

Comment:
there are few fields mentioning JSON but they are still byte[], I would expect maybe not a JSON parsed object but maybe a string there, a valid Json string should fit a C# string, maybe you don't want to deal with invalid data, encoding, etc and prefer to expose raw data?

Reply:
The native client is already equipped with a json parser, so any additional invalid data detection etc. on the C# side would be redundant. Given the fact that the documentation contains a link to the specification containing instructions on how the JSON should be serialized, I think it should be acceptable to simply expose the raw byte array to the user.